[GStreamer] Fix applying video decoder limit in unified build

blino opened the pull request:
#### 85e79c9134259f81bcd386299e15bdf0e67dfb26
<pre>
[GStreamer] Fix applying video decoder limit in unified build
<a href="https://bugs.webkit.org/show_bug.cgi?id=264878">https://bugs.webkit.org/show_bug.cgi?id=264878</a>

Reviewed by Michael Catanzaro.

* Source/WebCore/platform/SourcesGStreamer.txt:
If GStreamerRegistryScanner.cpp is built in unified files, the local
COMPILE_DEFINITIONS are not applied, so VIDEO_DECODING_LIMIT is
ineffective. Apply the definitions to the whole WebCore instead.

Canonical link: <a href="https://commits.webkit.org/270842@main">https://commits.webkit.org/270842@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/88411a512b881779e9d6e9e41c92c630400f6623

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26539 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5154 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27786 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28750 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24270 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26948 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6960 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2562 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24233 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26799 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3982 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22800 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3505 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3551 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23786 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29233 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24195 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24184 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29820 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3584 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1787 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27714 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5020 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6380 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4058 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3912 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
If this was hard to fix any other way, I would approve this, but I'm going to suggest the definition should be set globally instead. That's easy enough to do. Restricting the definition to just one source file is just not a good enough reason to remove a file from the unified build.

calvaris reviewed (changes requested):
Well, I agree with Michael here. 

> I can do this, but there are already plenty of `@no-unify` files in `Source/WebCore/platform/SourcesGStreamer.txt`, including `platform/audio/gstreamer/WebKitWebAudioSourceGStreamer.cpp` for the `GLIB_DISABLE_DEPRECATION_WARNINGS=1` definition.

Regarding this. It is different. One is to hide a build error, which is a footgun and the other one is enabling deprecated APIs.

blino committed:
[GStreamer] Fix applying video decoder limit in unified build
https://bugs.webkit.org/show_bug.cgi?id=264878

Reviewed by Michael Catanzaro.

* Source/WebCore/platform/SourcesGStreamer.txt:
If GStreamerRegistryScanner.cpp is built in unified files, the local
COMPILE_DEFINITIONS are not applied, so VIDEO_DECODING_LIMIT is
ineffective. Apply the definitions to the whole WebCore instead.

Canonical link: https://commits.webkit.org/270842@main
